transactions.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"hackathon-example/config"
+	"log"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func searchLatestTransactions(cfg config.Config, client *ethclient.Client, lastBlockNumber *uint64) {
+	blockNumber, err := client.BlockNumber(context.Background())
+	if err != nil {
+		log.Printf("Failed to get the latest block number: %v", err)
+		return
+	}
+
+	if blockNumber == *lastBlockNumber {
+		return
+	}
+
+	block, err := client.BlockByNumber(context.Background(), big.NewInt(int64(blockNumber)))
+	if err != nil {
+		log.Printf("Failed to get block: %v", err)
+		return
+	}
+
+	log.Printf("Searching for transaction at last processed block number: %d\n", blockNumber)
+	for i, tx := range block.Transactions() {
+		if i == 0 {
```

Comment:
Why are we skipping the first transaction in a block?

Reply:
The first transaction is always an empty one. Index of txs start at 1.

Reply:
Can you add a comment on this in the code?

Reply:
Yes for sure, good idea.